src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -126,19 +135,29 @@ trait RegularSync {
         //we have same chain prefix
         if (parent.hash == headers.head.parentHash) {
 
-          val currentBranchTotalDifficulty = headersQueue.map(_.number)
+          val oldBranch = headersQueue.map(_.number)
             .map(blockNumber => blockchain.getBlockByNumber(blockNumber))
-            .collect{
-              case Some(b) => b.header.difficulty
-              case None => BigInt(0)
-            }.sum
+
+          val oldBlocks: Seq[(BlockBody, BigInt)] = oldBranch
+            .collect { case Some(b) => b }
+            .map{b => (b.body, b.header.difficulty)}
```

Comment:
> adamsmo 4 hours ago Member
> for me it is more convenient

It's not a matter of convenience. It's about performance - `getBlockByNumber` is a DB call, is it not?